cmd/client-daemon.go

```diff
@@ -103,33 +103,87 @@ var clientDaemonCmd = &cobra.Command{
 	},
 }
 
+func (c *Client) tryStartJob() error {
+	var task string = c.config.Client.Task
+	// 5 attempts arbitrarily chosen - up to the orchestrator to send the correct task
+	var err error
+	for i := 0; i < 5; i++ {
+		pid, err := c.runTask(task)
+		if err == nil {
+			c.logger.Info().Msgf("managing process with pid %d", pid)
+			break
+		} else {
+			// enter a failure state, where we wait indefinitely for a command from NATS instead of
+			// continuing
+			c.logger.Info().Msgf("failed to run task with error: %v, attempt %d", err, i+1)
+			c.state.Flag = cedana.JobStartupFailed
+			recoveryCmd := c.enterDoomLoop()
+			task = recoveryCmd.UpdatedTask
+		}
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) runTask(task string) (int32, error) {
+	var pid int32
+
+	if task == "" {
+		return 0, fmt.Errorf("could not find task in config")
+	}
+
+	// validation of the task happens at the server level,
+	// so there's no real concerns here about executing the task without proper validation
+
+	// we want the task to run detached so we can checkpoint it
+	cmd := exec.Command("/bin/sh", "-c", task)
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Setsid: true,
+	}
+
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
```

Comment:
Do we not need to throw stdin & stdout into /dev/null?

Reply:
yes we do, good catch

Reply:
fixed!